### DISCUSSION TITLE
We should probably remove idom.run()

### DISCUSSION THREAD

**Discussion by Archmonger** [idom-team/idom#654] [Ideas]
Or at the very least, move it into a subfolder that makes it clear that it is a development utility. It shouldn't be the primary advertised way of using IDOM. 

Using a development webserver for production use is a big no-no, and by prominently displaying `idom.run()` as the first steps to using IDOM shows we are encouraging people to do just that. I believe the primary way of utilizing IDOM should be embedding within existing applications (see https://github.com/idom-team/idom/issues/653).

---

If we really want a `run()` callable to exist as a shortcut for running a simple single component application, we can consider is bundling a cross-platform ASGI webserver within this project and have `idom.run()` utilize that instead of running a development webserver. [Hypercorn](https://github.com/pgjones/hypercorn) and [Daphne](https://pypi.org/project/daphne/) are the only ASGI webserver I've tested be fully cross platform with no issues.

But even in this scenario I would still recommend changing the callable path to something like `idom.shortcuts.run()`.

**Comment by rmorshea** [comment]
Good point. Sanic (the default) runs with a production server but the others don't.
